ibm-ups/ups.cpp

```diff
@@ -71,8 +74,23 @@ UPS::UPS(sdbusplus::bus::bus& bus) :
     bool skipSignals{true};
     initializeDBusProperties(skipSignals);
 
-    // Now emit D-Bus signal that object has been created
+    // Read from cable sufficient number of times to determine actual UPS status
+    for (int i = 1; i <= (requiredMatchingReadCount + 1); ++i)
+    {
+        refresh();
+        using namespace std::chrono_literals;
+        std::this_thread::sleep_for(50ms);
+    }
+
+    // Emit D-Bus signal that object has been created
     emit_object_added();
+
+    // Force PropertiesChanged events to be emitted for the three UPS status
```

Comment:
I would say those appllications should be fixed to watch for interfaces added.  This is the first time I've ever seen the d-bus server work around a client not doing the right thing...

Reply:
Andrew mentioned preferring not to add an InterfacesAdded listener to the chassis state manager application.  I'd also need to get changes into PLDM for the same issue.

Given my lack of experience changing those two repos and the current release schedules, I thought this was a less risky approach.

If you think I need to change those other two applications instead I can pursue that.

Reply:
Given where we are, etc, I guess I'm OK with this.